File: ignite-nest/src/post/infrastructure/repository/post-command.adapter.ts
import { Injectable } from '@nestjs/common';
import { PostOutPort } from 'src/post/domain/port/out/post-out.port';
import { Post } from 'src/post/entity/post.entity';
import { PostUploadDto } from '../web/dto/request/post-upload.dto';
import { Repository } from 'typeorm';
import { PostMapper } from 'src/post/mapper/post.mapper';
import { InjectRepository } from '@nestjs/typeorm';
import { PostUpdateDto } from '../web/dto/request/post-update.dto';

@Injectable()
export class PostCommandAdapter implements PostOutPort {
  constructor(
    @InjectRepository(Post)
    private readonly postRepository: Repository<Post>,
  ) {}

  async save(postUploadDto: PostUploadDto): Promise<Post> {
    const post = PostMapper.toEntity(postUploadDto);
    return this.postRepository.save(post);
  }

  async update(post: Post, postUploadDto: PostUpdateDto): Promise<Post> {
    post.update(postUploadDto.title, postUploadDto.content);
    return await this.postRepository.save(post);
  }

  async delete(post: Post): Promise<void> {
    await this.postRepository.remove(post);
  }

  async updateCommentCount(post: Post, value: number): Promise<void> {
    post.updateCommentCount(value);
    await this.postRepository.save(post);
  }
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/post/application/adapter/PostCommandAdapter.kt
package org.rookedsysc.ignitespring.post.application.adapter

import org.rookedsysc.ignitespring.post.domain.port.out.PostOutPort
import org.rookedsysc.ignitespring.post.entity.PostEntity
import org.rookedsysc.ignitespring.post.infrastructure.repository.PostRepository
import org.springframework.stereotype.Repository

@Repository
class PostCommandAdapter(
    private val postRepository: PostRepository,
): PostOutPort {

    override fun save(post: PostEntity): PostEntity {
        return postRepository.save(post)
    }
}
